platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/network/topology/ConnectivityManager.java

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright (C) 2022-2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.platform.network.topology;
+
+import static com.swirlds.logging.legacy.LogMarker.EXCEPTION;
+import static com.swirlds.logging.legacy.LogMarker.NETWORK;
+
+import com.swirlds.common.platform.NodeId;
+import com.swirlds.platform.network.Connection;
+import com.swirlds.platform.network.ConnectionManager;
+import com.swirlds.platform.network.InboundConnectionManager;
+import com.swirlds.platform.network.OutboundConnectionManager;
+import com.swirlds.platform.network.PeerInfo;
+import com.swirlds.platform.network.connectivity.OutboundConnectionCreator;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/**
+ * Pre-builds connection managers for the supplied topology, does not allow changes at runtime
+ */
+public class ConnectivityManager {
+    private static final Logger logger = LogManager.getLogger(ConnectivityManager.class);
+    private final NetworkTopology topology;
```

Comment:
topology seems to be unchanged even when we update peers, which seems to indicate that updatePeers will not work. if unit tests did not catch this, then they need to be expanded.

Reply:
Hmm this isn't clear to me yet. Is there a requirement for the topology to change in this class? Always understood the connectivity manager's responsibility begins *after* the network topology has already been decided, with that topology passed down here. Are you thinking for DAB, we have to change the topology here? If yes, It's not something we've discussed but happy to hash it out and document.

Reply:
updating peers is a change in topology, it cannot remain the same

Reply:
Got it. Can you expand further on what should be changed in the topology and why? Happy to hop a quick call if that works better.

Reply:
sure, ping me on slack and we can arrange a call